### ISSUE TITLE
Increment Build Number Git

### ISSUE THREAD

**Issue by KrauseFx** [KrauseFx/fastlane#35]
Right now, the version bump is not commited and pushed.

How can we improve this? One solution is to use the Jenkins Build Number. The problem with this is when you migrate to a new Jenkins, it will be lower than the previous build.

**Comment by litso** [comment]
What are the cases where you wouldn't want the build number committed to the repo?

**Comment by KrauseFx** [comment]
I always want this commited, but what happens if the Jenkins job is run for each commit? Also, we can't just commit all changed files, since some setups may modify the project before building the app.

What do you suggest?

**Comment by bassrock** [comment]
You could keep the version number in some sort of lock file that specifies the commit it was built for. Then when you re commit you only need to commit that lock file. Also fast lane could compare the commit and only build and increment if it was more than the lock file changed?

**Comment by KrauseFx** [comment]
This is implemented with the latest release

**Comment by bassrock** [comment]
Awesome. Is there a doc somewhere that describes how to use it with a Jenkins setup? Maybe I'm missing something

**Comment by KrauseFx** [comment]
@bassrock do you mean something like https://github.com/KrauseFx/fastlane/blob/master/Jenkins.md? 

Or how to setup version commits and the other things?

**Comment by bassrock** [comment]
@KrauseFx yea I already have it setup like that. I mean jenkins setup with the version commit bumps, without running another build for a version bump.

**Comment by KrauseFx** [comment]
Honestly, I haven't used it myself, but @lmirosevic did probably. 

I want to provide a more complex example setup of `fastlane`, not sure when I get the time to do so.

**Comment by seanoshea** [comment]
I've used PListBuddy in the past to drag a build number out of Jenkins. https://medium.com/salesforce-ux/infograin-plistbuddy-is-your-buddy-b969edb642ef has a sample run script that you could add to your Jenkins specific target?

**Comment by KrauseFx** [comment]
@seanoshea It's not a good idea to use the Jenkins build number, as the number might get reset when you migrate to a new build machine.

**Comment by seanoshea** [comment]
@KrauseFx I believe Jenkins has the ability of re-setting the initial build number which would help if you moved to a new build machine. Not exactly ideal, but would be one way around it.

**Comment by bassrock** [comment]
This may be an answer we are looking for: https://github.com/jenkinsci/ci-skip-plugin

Just need to enable this and then make the commit message for a version bump include: "[ci skip]"

**Comment by KrauseFx** [comment]
That's a pretty neat idea :+1:

**Comment by ctrevarthen** [comment]
Unfortunately, if your CI job is listening for pushes to a branch, the Jenkins ci-skip-plugin runs after the fact and still requires Jenkins to kick off a build to determine if it should ignore the commit. The job then cancels, leaving the status unknown until a "real" commit comes through.

Instead of the ci-skip-plugin, we configured our Jenkins job's Source Code Management->Additional Behaviors to add the "Polling ignores commits from certain users" behavior. We added the CI "jenkins" user to that list, so any commits coming from the CI server (assuming we're only bumping the version number) will be ignored.
